client/src/types/client.rs

```diff
@@ -192,9 +192,11 @@ impl Client {
     /// Loads the state of [`Self`] from a [`ClientState`]. Replaces all previous data.
     ///
     /// # Example
-    pub(crate) fn restore(&self, state: ClientState, id: ClientId) -> Result<(), ClientError> {
+    pub(crate) fn restore(&mut self, state: ClientState, id: ClientId) -> Result<(), ClientError> {
```

Comment:
Hmm. I think it's better to put `Clientid` into a `Mutex` and keep the external immutability.